util/keyfile/src/lib.rs

```diff
@@ -53,14 +53,36 @@ pub fn read_root_entropy_keyfile_data<R: Read>(buffer: R) -> Result<RootIdentity
     Ok(serde_json::from_reader::<R, RootIdentityJson>(buffer)?.into())
 }
 
-/// Read user root identity from disk
+/// Read user mnemonic from disk
+pub fn read_mnemonic_keyfile<P: AsRef<Path>>(path: P) -> Result<AccountKey, Error> {
+    read_mnemonic_keyfile_data(File::open(path)?)
+}
+
+/// Read user root identity from any implementor of `Read`
+pub fn read_mnemonic_keyfile_data<R: Read>(buffer: R) -> Result<AccountKey, Error> {
+    Ok(serde_json::from_reader::<R, UncheckedMnemonicAccount>(buffer)?.try_into()?)
+}
+
+/// Read an account either in the RootIdentity format or the mnemonic format
+/// from disk
 pub fn read_keyfile<P: AsRef<Path>>(path: P) -> Result<AccountKey, Error> {
     read_keyfile_data(File::open(path)?)
 }
 
-/// Read user root identity from any implementor of `Read`
+/// Read an account key file in either format
 pub fn read_keyfile_data<R: Read>(buffer: R) -> Result<AccountKey, Error> {
-    Ok(serde_json::from_reader::<R, UncheckedMnemonicAccount>(buffer)?.try_into()?)
+    let value = serde_json::from_reader::<R, serde_json::Value>(buffer)?;
```

Comment:
Should this function be calling read mnemonic_keyfile_data and read_root_entropy_keyfile_data?

Reply:
Not in my opinion, it could simply be, read any keydata, and if more are created they can be added

Reply:
sorry misread your comment,

I think it's simpler to parse to json first, inspect it, then decide whether to parse the json one way or the other. because you can't "reset" a `Read` buffer afaik

Reply:
i mean this is basically the same way we made the python code work